contracts/Loan.sol

```diff
@@ -246,7 +239,7 @@ contract Loan is FDT, Pausable {
         _whenProtocolNotPaused();
         _isValidState(State.Active);
         (uint256 total, uint256 principal, uint256 interest,, bool paymentLate) = getNextPayment();
-        paymentsRemaining--;
+        --paymentsRemaining;
```

Comment:
If I am not wrong this will update the `paymentRemaining` at last, I am making a guess from the C knowledge I have. And what is the rationale of this change?

Reply:
`++x` and `--x` are immediate, where as `x++` or `x--` keeps x on the stack to be used for ops (nothing in this case) and then increments or decrements. I have done a good amount of testing with this and, because solc is perfect, I have noticed its always best to `--x` or `++x` if you don't need the post-decrement/increments effects.